Sources/Sentry/SentryTracer.m

```diff
@@ -425,10 +425,16 @@ - (void)finishInternal
 {
     [self cancelDeadlineTimer];
     if (self.isFinished) {
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+        discardProfilerForTracer(self);
```

Comment:
Not a bad idea! Let me try it out.

Reply:
The one wrinkle here is that a tracer's start timestamp can be changed, is that still true?

Reply:
The other issue with this is that timed out tracers won't be detected until another tracer starts, if that's where we do the cleanup search. If another one is never started, then we'll have a dangling pointer. I guess having that upper bound is better than infinity, though.

Reply:
Hmm, I also just noticed that the 500s limit is only a duration limit enforced after computation. Automatic tracers actually also have a 30 second timeout. But this is only for automatic tracers, not manually created ones. If there is no time limit for these, then we can't simply track timestamps.

Reply:
Manual transactions don't have a duration limit. Users could keep it alive for hours. So my suggestion won't work.

I thought about it and I think we only have one problem to solve with the current solution: When a user starts a transaction, never calls finish, and doesn't keep a reference to the transactions, `SentryProfiledTracerConcurrency` will keep a reference to the profile and the traceID forever. The `dealloc` of the SentryTracer could call `discardProfilerForTracer`. `dealloc` would also be our insurance if we forget to call `discardProfilerForTracer` somewhere.

Furthermore, I think we only need to call `discardProfilerForTracer` here https://github.com/getsentry/sentry-cocoa/blob/43860453150ea689c604b2a67d940dc675b70cdd/Sources/Sentry/SentryTracer.m#L442

We use this callback also here https://github.com/getsentry/sentry-cocoa/blob/43860453150ea689c604b2a67d940dc675b70cdd/Sources/Sentry/SentryPerformanceTracker.m#L233-L238

We don't need to consider transactions that never finish in the `SentryPerformanceTracker` as it only keeps track of automatic transactions and they have a timeout.

Doing those two suggested changes would do the job, I believe.

Please add some tests after doing the changes here.

Reply:
I think this will be fine, yes. At the point where there is any memory growth due to unended transactions, that's a logic error by the SDK consumer, not the SDK itself.

Reply:
I think just adding the call to `discardProfilerForTracer` in `SentryTracer.dealloc` will be sufficient. We can't call that before/after this https://github.com/getsentry/sentry-cocoa/blob/43860453150ea689c604b2a67d940dc675b70cdd/Sources/Sentry/SentryTracer.m#L442 

because then the profiler will be gone by the time we try to get its data later in `finishInternal` at the call to `captureTransactionWithProfile` (which eventually goes on to remove the tracking for the tracer, that's the happy path).

Reply:
I added several new tests, please check them out!

Reply:
I guess by logic error I just meant, if they start tons of endless transactions _and_ keep refs to them, _but also_ expect our memory to not grow accordingly.